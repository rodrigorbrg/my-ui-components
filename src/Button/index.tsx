export { default } from './button_view';
