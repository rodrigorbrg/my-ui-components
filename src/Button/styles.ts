import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: 12,
    textAlign: 'center',
  },
});
