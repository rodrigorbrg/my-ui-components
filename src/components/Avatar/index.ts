export { default } from './avatar_view';
