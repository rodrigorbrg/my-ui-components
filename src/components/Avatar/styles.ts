import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    height: 60,
    width: 60,
    marginVertical: 8,
    marginHorizontal: 1,
  },
  avatar: {
    borderRadius: 25,
    borderWidth: 1,
    justifyContent: 'center',
  },
  label: {
    textAlign: 'center',
    fontSize: 12,
  },
});
