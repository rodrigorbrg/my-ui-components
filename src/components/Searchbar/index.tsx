export { default } from './search_view';
