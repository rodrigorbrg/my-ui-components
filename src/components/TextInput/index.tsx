export { default } from './text_input_view';
