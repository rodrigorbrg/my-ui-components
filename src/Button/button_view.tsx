import React, { useCallback } from 'react';
import { View, TouchableOpacity, ActivityIndicator } from 'react-native';

import styles from './styles';

export default function Button({
  onPress,
  children,
  color,
  mode,
  loading,
  disabled,
}: {
  onPress?: () => void;
  children: React.ReactNode;
  color?: 'primary' | 'secondary';
  mode?: 'full' | 'outlined' | 'text';
  loading?: boolean;
  disabled?: boolean;
}) {
  const styleButton = useCallback(() => {
    let backgroundColor = 'transparent';
    let borderColor = 'transparent';
    let textColor: string | null = null;

    switch (color) {
      case 'primary':
        backgroundColor = '#FF0000';
        break;
      case 'secondary':
        backgroundColor = '#00FF00';
        break;
      default:
        backgroundColor = '#00FF00';
        break;
    }

    switch (mode) {
      case 'full':
        backgroundColor = backgroundColor;
        borderColor = backgroundColor;
        textColor = '#FFF';
        break;
      case 'outlined':
        backgroundColor = 'transparent';
        borderColor = backgroundColor;
        textColor = backgroundColor;
        break;
      case 'text':
        backgroundColor = 'transparent';
        borderColor = '#FFF';
        textColor = backgroundColor;
        break;
      default:
        backgroundColor = backgroundColor;
        borderColor = backgroundColor;
        textColor = '#FFF';
        break;
    }

    //const childrenWithProps =
    React.Children.map(children, (child) => {
      if (React.isValidElement(child)) {
        if (child?.props.style) {
          return React.cloneElement(child, {
            style: [...child?.props?.style, { color: textColor }],
          });
        } else {
          return React.cloneElement(child, {
            style: { color: textColor },
          });
        }
      }
      return child;
    });

    return {
      backgroundColor,
      borderColor,
      borderRadius: 6,
    };
  }, [children, color, mode]);

  if (loading) {
    return (
      <View style={[styles.container]}>
        <ActivityIndicator size="small" color={''} />
      </View>
    );
  }

  return (
    <View style={styles.container}>
      <TouchableOpacity
        style={styleButton()}
        onPress={onPress}
        disabled={disabled}
      >
        {children}
      </TouchableOpacity>
    </View>
  );
}
