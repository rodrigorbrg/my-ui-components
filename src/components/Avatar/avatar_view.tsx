import React from 'react';
import {
  ImageSourcePropType,
  Text,
  Pressable,
  Image,
  View,
} from 'react-native';

import { withTheme, Theme } from '../../context/ThemeProvider';
import styles from './styles';

function Avatar({
  img,
  label,
  onPress,
  theme,
}: {
  img?: ImageSourcePropType;
  label?: string;
  onPress: () => void;
  theme: Theme;
}) {
  const { colors } = theme;
  return (
    <Pressable
      style={({ pressed }) => [
        styles.container,
        { opacity: pressed ? 0.7 : 1.0 },
      ]}
      onPress={onPress}
    >
      {img ? (
        <Image
          source={img}
          style={[styles.avatar, { borderColor: colors.surface_secondary }]}
        />
      ) : (
        <View
          style={[
            styles.avatar,
            // eslint-disable-next-line react-native/no-inline-styles
            {
              height: 40,
              width: 40,
              backgroundColor: colors.surface_secondary,
            },
          ]}
        />
      )}
      {label && (
        <Text
          style={[
            styles.label,
            {
              color: colors.text_on_background,
            },
          ]}
        >
          {label}
        </Text>
      )}
    </Pressable>
  );
}

export default withTheme(Avatar);
