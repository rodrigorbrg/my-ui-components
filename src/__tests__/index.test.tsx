// it.todo('write a test');
import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import Avatar from '../components/Avatar';
import Button from '../components/Button';
import Searchbar from '../components/Searchbar';
import TextInput from '../components/TextInput';

const submitButton = jest.fn();

describe('First test', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('Button render correctly', async () => {
    const { queryByText } = render(
      <Button onPress={submitButton}>{'Button Test'}</Button>
    );

    const instance = queryByText('Button Test');
    expect(instance).not.toBeNull();

    fireEvent.press(instance);

    expect(submitButton).toHaveBeenCalledTimes(1);
  });

  it('Avatar render correctly', async () => {
    const avatar = {
      key: '1',
      img: require('../assets/thought.png'),
      label: 'Rock',
      onPress: submitButton,
    };
    const { queryByText } = render(<Avatar {...avatar} />);

    const instance = queryByText('Rock');
    expect(instance).not.toBeNull();

    fireEvent.press(instance);

    expect(submitButton).toHaveBeenCalledTimes(1);
  });

  it('Searchbar render correctly', async () => {
    const mockSetState = jest.fn();
    const useStateMock: any = (initState: any) => [initState, mockSetState];

    jest.spyOn(React, 'useState').mockImplementation(useStateMock);

    const [state, setState] = useStateMock('');

    const { queryByPlaceholderText } = render(
      <Searchbar
        value={state}
        onChange={setState}
        placeholder={'Busque aqui!'}
      />
    );

    const instance = queryByPlaceholderText('Busque aqui!');
    expect(instance).not.toBeNull();

    fireEvent.changeText(instance, 'bandas');
    expect(mockSetState).toHaveBeenCalledWith('bandas');
  });

  it('TextInput render correctly', async () => {
    const mockSetState = jest.fn();
    const useStateMock: any = (initState: any) => [initState, mockSetState];

    jest.spyOn(React, 'useState').mockImplementation(useStateMock);

    const [state, setState] = useStateMock('');

    const { queryByPlaceholderText } = render(
      <TextInput
        value={state}
        onChangeText={setState}
        placeholder={'Texto aqui!'}
      />
    );

    const instance = queryByPlaceholderText('Texto aqui!');
    expect(instance).not.toBeNull();

    fireEvent.changeText(instance, '111.111.111-11');
    expect(mockSetState).toHaveBeenCalledWith('111.111.111-11');
  });
});
