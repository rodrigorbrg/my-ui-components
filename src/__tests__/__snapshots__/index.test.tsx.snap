// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First test Button is disable correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "opacity": 0.5,
      },
      Object {
        "backgroundColor": "#29b6f6",
        "borderColor": "#29b6f6",
        "borderRadius": 6,
        "borderWidth": 1,
        "height": 40,
        "opacity": 0.5,
      },
      Object {
        "justifyContent": "center",
        "paddingHorizontal": 8,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "textAlign": "center",
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  >
    Button Test
  </Text>
</View>
`;

exports[`First test Button render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "opacity": 1,
      },
      Object {
        "backgroundColor": "#29b6f6",
        "borderColor": "#29b6f6",
        "borderRadius": 6,
        "borderWidth": 1,
        "height": 40,
        "opacity": 1,
      },
      Object {
        "justifyContent": "center",
        "paddingHorizontal": 8,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "textAlign": "center",
        },
        Object {
          "color": "#FFF",
        },
      ]
    }
  >
    Button Test
  </Text>
</View>
`;
